core/commands/storage/session.go

```diff
@@ -358,21 +360,22 @@ func (ss *FileContracts) GetOrDefault(hash string, shardSize uint64, length int6
 		c.RetryChan = make(chan *StepRetryChan)
 		c.StartTime = time.Now()
 		c.State = InitState
-		c.Size = shardSize
-		c.Length = time.Duration(length*24) * time.Hour
+		c.ShardSize = shardSize
+		c.StorageLength = length
+		c.ContractLength = time.Duration(length*24) * time.Hour
 		c.Price = price
-		c.TotalPay = price * int64(shardSize) * length
+		c.TotalPay = price * int64(shardSize) * length / int64(units.GiB)
```

Comment:
Two issues: overflow and precision loss.

Try `int64(float64(price) * (float64(shardSize) / float64(units.GB)) * float64(length))`